### PR TITLE
[ASM] - Expandr 4735

### PR DESCRIPTION
**Pull request by kball-pa**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://jira-hq.paloaltonetworks.local/browse/EXPANDR-4735

## Description
Currently, we score and return all owners in ${alert.asmserviceowner} in sorted order. Some of these may be service accounts or other low-confidence users/owners that we don't want to notify. 

This PR implements a ranking algorithm for Service Ownership that tries to find a smaller (targeting roughly ~5), high-confidence set of owners that we would be comfortable notifying via email. After the Service Ownership playbook runs, ${alert.asmserviceowner} will contain this smaller, high-confidence set, while ${alert.asmserviceownerunrankedraw} will contain the full set of (deduplicated) owners pulled during enrichment.

See unit tests for detailed specification for how the ranking algorithm works.

Test plan: pytest + manual testing in tenant (see JIRA ticket)

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ x] No

## Must have
- [ x] Tests
- [ x] Documentation

### CODE REVIEW

**Review by johnnywilkes — APPROVED** [review]
@kball-pa , Approved.  Nice work!

@michal-dagan , please merge when possible

File: Packs/CortexAttackSurfaceManagement/Scripts/RankServiceOwners/RankServiceOwners.py [hunk 1]
```diff
@@ -139,6 +144,54 @@ def aggregate(owners: List[Dict[str, str]]) -> List[Dict[str, Any]]:
     return deduped
 
 
+def _get_k(
+    scores: Iterable[float],
+    target_k: int = 5,
+    k_tol: int = 2,
+    a_tol: float = 1.0,
+    min_score_proportion: float = 0.75
```
Comment (ptoman-pa) [inline Packs/CortexAttackSurfaceManagement/Scripts/RankServiceOwners/RankServiceOwners.py:152]: can you defensively verify that these values are within expected bounds? (for instance, that k_tol isn't -3.)

File: Packs/CortexAttackSurfaceManagement/Scripts/RankServiceOwners/RankServiceOwners_test.py [hunk 2]
```diff
@@ -395,3 +451,56 @@ def test_main(mocker, owners, expected_out, capfd):
     # Verify the output value was set
     expected_calls_to_mock_object = [unittest.mock.call('setAlert', {'asmserviceowner': expected_out})]
     assert demisto_execution_mock.call_args_list == expected_calls_to_mock_object
+
+
+def test_get_k():
+    """
+    These cases are designed to specify the intuition we are trying to implement with the algorithm.
+    They are specific to a target value of 5; if the target_k changes, these tests should update to reflect that.
```
Comment (ptoman-pa) [inline Packs/CortexAttackSurfaceManagement/Scripts/RankServiceOwners/RankServiceOwners_test.py:459]: The docstring should be generic, so mentioning 5 is not appropriate here unless you're actually setting that parameter in _get_k -- instead you might mention that this test verifies the default case.  (The place to note "5" is in the docstring for _get_k -- there it is worth flagging that all the default values are tuned for k=5.)

**Review by ptoman-pa — APPROVED** [review]

**Comment by michal-dagan** [comment]
Hi @kball-pa,
Thank you for your contribution!
Good work :)
1. please run  `demisto-sdk pre-commit -g --unit-test --validate --no-secrets --show-diff-on-failure --verbose` in order to pass the checks.
2. Please add a unit-test as the build fails on [test-coverage](https://app.circleci.com/pipelines/github/demisto/content/273326/workflows/e2f08fa4-8e07-4035-97f0-b7ddfc6f2758/jobs/706881/parallel-runs/0/steps/0-121):
Message: "[red]Unit-tests for 'Packs/CortexAttackSurfaceManagement/Scripts/RankServiceOwners/RankServiceOwners.py' must reach a coverage of at least 95.05% (currently: 93.0%).[/red]"

Please feel free to reach out to me with any questions - I'm available here or on slack :)
Thanks again

**Comment by kball-pa** [comment]
@michal-dagan I increased the test coverage and verified that all checks were passing on a previous commit [5d6c938](https://github.com/demisto/content/pull/27624/commits/5d6c93873d1cf9fe754b2d07618f414b48073535), but now the `Run Validations` step is failing with an error that seems to be local to the build.

**Review by michal-dagan — APPROVED** [review]
Great!

**Pull request merged**